src/highdicom/sr/sop.py

```diff
@@ -180,67 +180,168 @@ def __init__(
+        for pre in previous_versions:
+            pre_instance_item = Dataset()
+            pre_instance_item.ReferencedSOPClassUID = pre.SOPClassUID
+            pre_instance_item.ReferencedSOPInstanceUID = pre.SOPInstanceUID
+            key = (pre.StudyInstanceUID, pre.SeriesInstanceUID)
+            collection[key].append(pre_instance_item)
+        return self._create_references(collection)
+
+    def _collect_evidence(
+        self,
+        evidence: List[Dataset],
+        content: Dataset
+    ) -> Tuple[List[Dataset], List[Dataset]]:
+        """Collect evidence for the SR document.
+
+        Any `evidence` that is referenced in `content` via IMAGE or
+        COMPOSITE content items will be grouped together for inclusion in the
+        Current Requested Procedure Evidence Sequence and all remaining
+        evidence will be grouped for potential inclusion in the Pertinent Other
+        Evidence Sequence.
+
+        Parameters
+        ----------
+        evidence: List[pydicom.dataset.Dataset]
+            Metadata of instances that serve as evidence for the SR document
+            content
+        content: pydicom.dataset.Dataset
+            SR document content
+
+        Returns
+        -------
+        Tuple[List[pydicom.dataset.Dataset], List[pydicom.dataset.Dataset]]
+            Items of the Current Requested Procedure Evidence Sequence and the
+            Pertinent Other Evidence Sequence
+
+        Raises
+        ------
+        ValueError
+            When a SOP instance is referenced in `content` but not provided as
+            `evidence`
+
+        """  # noqa
+        references = find_content_items(
+            content,
+            value_type=ValueTypeValues.IMAGE,
+            recursive=True
+        )
+        references += find_content_items(
+            content,
+            value_type=ValueTypeValues.COMPOSITE,
+            recursive=True
+        )
+        ref_uids = [
+            ref.ReferencedSOPSequence[0].ReferencedSOPInstanceUID
+            for ref in references
+        ]
+        evd_uids = []
+        ref_collection: Dict[str, List[Dataset]] = defaultdict(list)
+        unref_collection: Dict[str, List[Dataset]] = defaultdict(list)
+        for evd in evidence:
+            if evd.StudyInstanceUID != evidence[0].StudyInstanceUID:
+                raise ValueError(
+                    'Data sets provided as evidence must all belong to the '
+                    'same study.'
+                )
+            evd_item = Dataset()
+            evd_item.ReferencedSOPClassUID = evd.SOPClassUID
+            evd_item.ReferencedSOPInstanceUID = evd.SOPInstanceUID
+            key = (evd.StudyInstanceUID, evd.SeriesInstanceUID)
+            if evd.SOPInstanceUID in ref_uids:
+                ref_collection[key].append(evd_item)
+            else:
+                unref_collection[key].append(evd_item)
+            evd_uids.append(evd.SOPInstanceUID)
```

Comment:
Is it worth checking that the item is not already in the sequence, in case the user passes duplicate items in `evidence`?

Reply:
Addressed with 8e6ad6259f7553cbd708bbc392d2b2f41df540ae